### PR TITLE
Stats: Adding empty state component for Referrers

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Related to https://github.com/Automattic/red-team/issues/63

## Proposed Changes

* adding new empty state for Referrers
* refactoring empty state actions to be shared across empty states
* extend empty card styling to stretch the main content

## Why are these changes being made?

* a part of the empty states project

## Testing Instructions

* go to the live branch
* apply `stats/empty-module-traffic` feature flag
* verify that a page with empty Referrers component works with and without the feature flag
* repeat for blogs with data

SCR-20240627-mowk

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by annacmc — APPROVED** [review]
works for me 👍

File: client/my-sites/stats/features/modules/stats-referrers/stats-referrers.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	);
+	const data = useSelector( ( state ) =>
+		getSiteStatsNormalizedData( state, siteId, statType, query )
+	) as [ id: number, label: string ]; // TODO: get post shape and share in an external type file.
+
+	return (
+		<>
+			{ /* This will be replaced with ghost loaders, fallback to the current implementation until then. */ }
+			{ requesting && <StatsModulePlaceholder isLoading={ requesting } /> }
+			{ ( ! data || ! data?.length ) && (
```
Comment (kangzj) [inline client/my-sites/stats/features/modules/stats-referrers/stats-referrers.tsx:52]: Minor suggestion - I think if we are using conditional chaining here, then `!data` could be omitted, couldn't it? The same applies the L76.

**Review by kangzj — APPROVED** [review]
Works well and look great! Love it 👍

File: client/my-sites/stats/features/modules/stats-referrers/stats-referrers.tsx (hunk 1 above)
Comment (grzegorz-cp) [reply client/my-sites/stats/features/modules/stats-referrers/stats-referrers.tsx:52]: I'll update this with a fix coming in a moment.

File: client/my-sites/stats/features/modules/stats-referrers/stats-referrers.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	query: string;
+	moduleStrings: {
+		title: string;
+		item: string;
+		value: string;
+		empty: string;
+	};
+};
+
+const StatsRefeeres: React.FC< StatsRefeeresProps > = ( {
```
Comment (dognose24) [inline client/my-sites/stats/features/modules/stats-referrers/stats-referrers.tsx:30]:
```suggestion
const StatsReferrers: React.FC< StatsRefeeresProps > = ( {
```

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16253591

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @grzegorz-cp for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**